srv/adapter/type.ts

```diff
@@ -45,4 +45,6 @@ export type AdapterProps = {
   isThirdParty?: boolean
 }
 
-export type ModelAdapter = (opts: AdapterProps) => AsyncGenerator<string | { error: any }>
+export type ModelAdapter = (
+  opts: AdapterProps
+) => AsyncGenerator<string | { partial: string } | { error: any }>
```

Comment:
Added `{partial:string}` as a separate yielded type to more easily avoid emitting useless `message-partials` for non-streaming adapters.